File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.java
import java.util.HashMap;
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

public class Solution {
    public int findLeastNumOfUniqueInts(int[] arr, int k) {
        HashMap<Integer, Integer> array = new HashMap<>();
        for (int num : arr) {
            array.put(num, array.getOrDefault(num, 0) + 1);
        }
        List<Integer> values = new ArrayList<>(array.values());
        Collections.sort(values);
        int c = 0;
        for (int i = 0; i < values.size(); i++) {
            if (k > values.get(i)) {
                k -= values.get(i);
                values.set(i, 0);
            } else {
                values.set(i, values.get(i) - k);
                k = 0;
            }
            if (values.get(i) != 0) {
                c++;
            }
        }
        return c;
    }
}

File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.py
from collections import Counter
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        array=Counter(arr)
        values=list(array.values())
        values.sort()
        c=0
        for i in range(len(values)):
            if(k>values[i]):
                k-=values[i]
                values[i]=0
            else:
                values[i]-=k
                k=0
            if values[i]!=0:
                c+=1
        return c
       
          